### PR TITLE
blending of floating windows

### PR DESCRIPTION
**Pull request by bfredl**
From the useless eyecandy department.

```
let x = nvim_open_win(0, v:true, {'relative': 'editor', 'row': 4, 'col': 4, 'width': 30, 'height': 3})
set floatblend=20 " window local option
 ```

Blending can additionally be overriden per highlight group (inside a float). This works:

```
hi PmenuSel blend=0
```

To disable blending of current item when `pumblend` is otherwise active.

### CODE REVIEW

**Comment by bfredl** [comment]
Update: 'floatblend' is now a numerical option. It works like 'pumblend', except that it is window-local.
`hi ... blend=` is still supported for overrides (it stores the blend level in the `HlEntry` table anyway). But if 'pumblend' or 'floatblend' is not set, blending is disabled for that overlay (`blend=` overrides take no effect).

**Comment by bfredl** [comment]
Sketched implementation of CursorLine blending. Use `hi CursorLine guibg=#FFFFFF blend=85` to tint all colors in the cursorline region 15% brighter, regardless if the underlying text has `guibg=` or not (and thus sidesteps the entire priority shenanigans). Though this is mostly unrelated from 'floatblend', so I might make it a separate PR.

File: src/nvim/options.lua [hunk 1]
```diff
@@ -819,6 +819,13 @@ return {
       varname='p_fixeol',
       defaults={if_true={vi=true}}
     },
+    {
+      full_name='floatblend', abbreviation='fb',
```
Comment (justinmk) [inline src/nvim/options.lua]: `winfloatblend` or even just `winblend` makes this discoverable by `:set win<tab>`. And `winblend` is analogous to `pumblend`.

**Comment by justinmk** [comment]
> Use `hi CursorLine guibg=#FFFFFF blend=85` to tint all colors in the cursorline region 15% brighter, regardless if the underlying text has `guibg=` or not

A good sentence for the docs. Closes https://github.com/neovim/neovim/issues/9800

**Pull request merged**